qdrant-landing/content/documentation/fastembed/fastembed-custom-models.md

```diff
@@ -0,0 +1,525 @@
+
+1. Create a new repository on Hugging Face
+2. Upload the following files:
+   - `model.onnx`
+   - `config.json` (containing model configuration)
+   - `preprocessor_config.json` (containing preprocessing parameters)
+   
+```bash
+# Example structure of your Hugging Face repo
+your-repo/
+├── model.onnx
+├── config.json
+└── preprocessor_config.json
+```
+
+## Using the Custom Models
+
+Now you can use your custom models with FastEmbed:
+
+```python
+
+# Call the necessary function to get the onnx version of the model
+convert_efficientnet_to_onnx()
+convert_blip_to_onnx()
+
+# Register custom models
+from fastembed.image.image_embedding import ImageEmbedding
+
+ImageEmbedding.EMBEDDINGS_REGISTRY.extend([
+    CustomEfficientNetONNXEmbedding,
+    CustomBlipONNXEmbedding
+])
+
+# Apply the initialization modification
+modify_image_embedding_init()
+
+# Now upload all the model related files to respective huggingface repository
+
+# Use EfficientNet embeddings
+efficientnet_embedder = ImageEmbedding(
+    model_name="efficientnet-b0-custom",
+    cache_dir="./efficientnet_custom"
+)
+
+# Use BLIP embeddings
+blip_embedder = ImageEmbedding(
+    model_name="blip-image-encoder",
+    cache_dir="./blip_custom"
+)
+
+# Generate embeddings
+image_paths = ["image1.jpg", "image2.jpg"]
+efficientnet_embeddings = list(efficientnet_embedder.embed(image_paths))
+blip_embeddings = list(blip_embedder.embed(image_paths))
+
+# View embeddings
+
+# Print details of efficientnet_embeddings
+for i, embedding in enumerate(efficientnet_embeddings):
+    print(f"Embedding {i + 1}:")
+    print(f"Shape: {embedding.shape}")
+    print(f"First 5 values: {embedding[:5]}")
+    
+# Print details of blip_embeddings
+for i, embedding in enumerate(blip_embeddings):
+    print(f"Embedding {i + 1}:")
+    print(f"Shape: {embedding.shape}")
+    print(f"First 5 values: {embedding[:5]}")
+
+```
+
+## Best Practices and Tips
+
+1. **Model Selection**: Choose models that best fit your specific use case. EfficientNet is great for general image features, while BLIP excels at understanding image-text relationships.
+
+2. **Performance Optimization**:
+   - Use batch processing when embedding multiple images
+   - Consider quantization for smaller model size
+   - Implement proper error handling for robustness
+
+3. **Memory Management**: Be mindful of batch sizes, especially when dealing with large images or multiple models.
+
+4. **Version Control**: Maintain proper versioning for your models and configurations on Hugging Face.
+
+
+## Practical Applications with Qdrant
+
+By leveraging Qdrant alongside FastEmbed, businesses can unlock:
+- **Intelligent Search**: Enable users to find relevant content faster.
+- **Enhanced Recommendations**: Deliver tailored suggestions in e-commerce, media, and more.
+- **Domain-Specific Insights**: Power unique applications in medicine, law, or academia.
+

```

Comment:
Our users rather understand why they want to achieve. If that's not for SEO, I would remove this part.